stdlib/src/memory/arc.mojo

```diff
@@ -162,3 +162,11 @@ struct Arc[T: Movable](CollectionElement, CollectionElementNew):
         """
         # TODO: consider removing this method.
         return UnsafePointer.address_of(self._inner[].payload)
+
+    fn count(self) -> Int64:
```

Comment:
Thoughts on `UInt` as the return type instead?  We know the number of references will always be non-negative.

Reply:
Conceptually I agree, but the datatype used in `_ArcInner` is a `DType.int64`, and I don't know if anything funny may happen during conversion. Do I just change it?

Reply:
Let's chase that type down after I land this, how's that sound?  It predates us having a `UInt` type, so that's why the inner type is the way it is I suspect.

Reply:
I changed it to UInt64 in the latest push hehe, theoretically this type  is just for internal counting... Do I revert the  change or do we see if it lands?

Reply:
!sync